core/src/main/java/tech/tablesaw/joining/DataFrameJoiner.java

```diff
@@ -85,142 +112,145 @@ public Table inner(Table table2, String col2Name) {
+                CategoricalColumn<?> column = columns[i];
+                CategoricalColumn<?> table1Column = table1.categoricalColumn(column.name());
+
+                ColumnType type = table1Column.type();
+                // relies on both arrays, columns, and col2Names,
+                // having corresponding values at same index
+                String col2Name = col2Names[i];
+                Selection rowBitMapOneCol = null;
+                if (type instanceof DateColumnType) {
+                    IntIndex index = new IntIndex(table2.dateColumn(col2Name));
+                    DateColumn col1 = (DateColumn) table1Column;
+                    int value = col1.getIntInternal(ri);
+                    rowBitMapOneCol = index.get(value);
+                } else if (type instanceof DateTimeColumnType) {
+                    LongIndex index = new LongIndex(table2.dateTimeColumn(col2Name));
+                    DateTimeColumn col1 = (DateTimeColumn) table1Column;
+                    long value = col1.getLongInternal(ri);
+                    rowBitMapOneCol = index.get(value);
+                } else if (type instanceof TimeColumnType) {
+                    IntIndex index = new IntIndex(table2.timeColumn(col2Name));
+                    TimeColumn col1 = (TimeColumn) table1Column;
+                    int value = col1.getIntInternal(ri);
+                    rowBitMapOneCol = index.get(value);
+                } else if (type instanceof StringColumnType) {
+                    StringIndex index = new StringIndex(table2.stringColumn(col2Name));
+                    StringColumn col1 = (StringColumn) table1Column;
+                    String value = col1.get(ri);
+                    rowBitMapOneCol = index.get(value);
+                } else if (type instanceof IntColumnType) {
+                    IntIndex index = new IntIndex(table2.intColumn(col2Name));
+                    IntColumn col1 = (IntColumn) table1Column;
+                    int value = col1.getInt(ri);
+                    rowBitMapOneCol = index.get(value);
+                } else if (type instanceof LongColumnType) {
+                    LongIndex index = new LongIndex(table2.longColumn(col2Name));
+                    LongColumn col1 = (LongColumn) table1Column;
+                    long value = col1.getLong(ri);
+                    rowBitMapOneCol = index.get(value);
+                } else if (type instanceof ShortColumnType) {
+                    ShortIndex index = new ShortIndex(table2.shortColumn(col2Name));
+                    ShortColumn col1 = (ShortColumn) table1Column;
+                    short value = col1.getShort(ri);
+                    rowBitMapOneCol = index.get(value);
                 } else {
-                    crossProduct(result, table1Rows, table2Rows);
+                    throw new IllegalArgumentException(
+                            "Joining is supported on numeric, string, and date-like columns. Column "
```

Comment:
I might just say categorical columns here because double columns are numeric, but not categorical, which makes this message a bit misleading

Reply:
If message was changed to say that only categorical columns are supported then BooleanColumn should already be accepted, but trying to join on a boolean column will currently filter to this else condition; I got this condition to be exercised with a junit test that joined on a boolean column.   BooleanColumn implements CategoricalColumn though so should an if condition be added here for BooleanColumnType?